File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/impl/RenameListInteractorImpl.java
package com.rachelleignacio.listbucket.domain.interactors.impl;

import com.rachelleignacio.listbucket.db.DbInteractor;
import com.rachelleignacio.listbucket.domain.executor.MainThread;
import com.rachelleignacio.listbucket.domain.executor.ThreadExecutor;
import com.rachelleignacio.listbucket.domain.interactors.base.AbstractInteractor;
import com.rachelleignacio.listbucket.domain.interactors.RenameListInteractor;
import com.rachelleignacio.listbucket.domain.models.List;

/**
 * Created by rachelleignacio on 4/18/17.
 */

public class RenameListInteractorImpl extends AbstractInteractor implements RenameListInteractor {
    private RenameListInteractor.Callback callback;
    private DbInteractor database;
    private List listToRename;
    private String newListName;

    public RenameListInteractorImpl(ThreadExecutor threadExecutor, MainThread mainThread, Callback callback,
                                    DbInteractor db, List listToRename, String newName) {
        super(threadExecutor, mainThread);
        this.callback = callback;
        this.database = db;
        this.listToRename = listToRename;
        this.newListName = newName;
    }

    @Override
    public void run() {
        database.renameList(listToRename, newListName);
        mainThread.post(new Runnable() {
            @Override
            public void run() {
                callback.onListRenamed();
            }
        });
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/impl/RenameListInteractorImpl.kt
package com.rachelleignacio.listbucket.domain.interactors.impl

import com.rachelleignacio.listbucket.db.DbInteractor
import com.rachelleignacio.listbucket.domain.executor.MainThread
import com.rachelleignacio.listbucket.domain.executor.ThreadExecutor
import com.rachelleignacio.listbucket.domain.interactors.RenameListInteractor
import com.rachelleignacio.listbucket.domain.interactors.base.AbstractInteractor
import com.rachelleignacio.listbucket.domain.models.List

/**
 * Created by rachelleignacio on 8/9/17.
 */
class RenameListInteractorImpl(threadExecutor: ThreadExecutor,
                               mainThread: MainThread,
                               private val callback: RenameListInteractor.Callback,
                               private val dbInteractor: DbInteractor,
                               private val listToRename: List,
                               private val newName: String)
    : AbstractInteractor(threadExecutor, mainThread), RenameListInteractor {

    override fun run() {
        dbInteractor.renameList(listToRename, newName)
        mainThread.post(Runnable { callback.onListRenamed() })
    }
}